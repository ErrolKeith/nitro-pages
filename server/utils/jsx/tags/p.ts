import { makeJsx } from "./makeJsx";

export interface ParagraphConfig {
  text: string;
  className?: string;
}

export function makeParagraph(config: ParagraphConfig) {
  const { text, className } = config;
  return makeJsx("p", { className }, text);
}
