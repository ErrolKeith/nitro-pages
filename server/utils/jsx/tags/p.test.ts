import { it, expect } from "vitest";
import { makeParagraph, ParagraphConfig } from "./p";

it("can make a p tag", () => {
  const pTestConig: ParagraphConfig = {
    text: "lorem ipsum",
  };
  const pTest = makeParagraph(pTestConig);
  expect(pTest).toMatchSnapshot();
});
