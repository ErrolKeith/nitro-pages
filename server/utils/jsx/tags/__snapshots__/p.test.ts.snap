// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make a p tag 1`] = `
<p>
  lorem ipsum
</p>
`;
